src/NemoStuff.jl

```diff
@@ -176,129 +176,6 @@ function sub(M::Generic.Mat, rows::AbstractUnitRange{Int}, cols::AbstractUnitRan
     return z
 end
 
-################################################################################
-#
-#  Concatenation of matrices
-#
-################################################################################
-
-@doc raw"""
-    vcat(A::Vector{Mat}) -> Mat
-
-Forms a big matrix by vertically concatenating the matrices in $A$.
-All component matrices need to have the same number of columns.
-"""
-function Base.vcat(A::Vector{T}) where {S<:RingElem,T<:MatElem{S}}
-    if any(x -> ncols(x) != ncols(A[1]), A)
-        error("Matrices must have same number of columns")
-    end
-    M = zero_matrix(base_ring(A[1]), sum(nrows, A), ncols(A[1]))
-    s = 0
-    for i = A
-        for j = 1:nrows(i)
-            for k = 1:ncols(i)
-                M[s+j, k] = i[j, k]
-            end
-        end
-        s += nrows(i)
-    end
-    return M
-end
-
-function Base.hcat(A::Vector{T}) where {S<:RingElem,T<:MatElem{S}}
-    if any(x -> nrows(x) != nrows(A[1]), A)
-        error("Matrices must have same number of rows")
-    end
-    M = zero_matrix(base_ring(A[1]), nrows(A[1]), sum(ncols, A))
-    s = 0
-    for i = A
-        for j = 1:ncols(i)
-            for k = 1:nrows(i)
-                M[k, s+j] = i[k, j]
-            end
-        end
-        s += ncols(i)
-    end
-    return M
-end
-
-function Base.hcat(A::MatElem...)
```

Comment:
Do we really need to remove this? I just tried with removing just the preceding methods, and `vcat(Union{}[])` as well as `vcat()` worked just fine.

Same question for the other methods taking `::MatElem...` args

Reply:
They already exist in better in `src/Matrix.jl`. So these removed methods here get never actually called